spec/controllers/cases/offender_sar_controller_spec.rb

```diff
@@ -68,6 +68,159 @@
     end
   end
 
+  describe '#create' do
+    before do
+      sign_in responder
+      post :create, params: params
+      expect(assigns(:case)).to be_a Case::SAR::Offender
+      expect(flash[:notice]).to eq nil
+    end
+
+    context 'partial validations' do
+      context 'for step subject-details' do
+        let(:params) do
+          {
+            current_step: 'subject-details',
+            offender_sar: {
+              subject_full_name: '',
+              subject_address: '',
+            }
+          }
+        end
+
+        it 'validates subject name and address' do
+          errors = assigns(:case).errors.messages
+          expect(errors[:subject_full_name]).to eq ["can't be blank"]
+          expect(errors[:subject_address]).to eq ["can't be blank"]
+        end
+
+        it 'sets empty values and validates' do
```

Comment:
I find this wording slightly confusing - is this not just testing that these things are validated? Or are you saying "sets" due to the controller work arounds for dates and things? :)

Reply:
thanks, had a go at clarifying as discussed, has it helped?